src/mame/misc/fruitctk.cpp

```diff
@@ -0,0 +1,316 @@
+	PORT_BIT( 0x20, IP_ACTIVE_LOW,  IPT_BUTTON1 ) PORT_CODE(KEYCODE_F) PORT_NAME("Win")     
+	PORT_BIT( 0x40, IP_ACTIVE_LOW,  IPT_GAMBLE_PAYOUT ) PORT_CODE(KEYCODE_G) // Payout?
+	PORT_BIT( 0x80, IP_ACTIVE_HIGH, IPT_COIN1 ) PORT_CODE(KEYCODE_H)// Coin. will cause error 30
+
+	PORT_START("Inputs3")
+	PORT_BIT( 0x01, IP_ACTIVE_HIGH, IPT_BUTTON1 ) PORT_CODE(KEYCODE_J)  
+	PORT_BIT( 0x02, IP_ACTIVE_HIGH, IPT_BUTTON1 ) PORT_CODE(KEYCODE_K)
+	PORT_BIT( 0x04, IP_ACTIVE_HIGH, IPT_BUTTON1 ) PORT_CODE(KEYCODE_L)
+	PORT_BIT( 0x08, IP_ACTIVE_HIGH, IPT_BUTTON1 ) PORT_CODE(KEYCODE_Z)
+	PORT_BIT( 0x10, IP_ACTIVE_HIGH, IPT_BUTTON1 ) PORT_CODE(KEYCODE_X)
+	PORT_BIT( 0x20, IP_ACTIVE_HIGH, IPT_BUTTON1 ) PORT_CODE(KEYCODE_C)
+	PORT_BIT( 0x40, IP_ACTIVE_HIGH, IPT_BUTTON1 ) PORT_CODE(KEYCODE_V)
+	PORT_BIT( 0x80, IP_ACTIVE_HIGH, IPT_BUTTON1 ) PORT_CODE(KEYCODE_B)
+
+	PORT_START("Inputs4")
+    PORT_BIT(0x01, IP_ACTIVE_LOW, IPT_UNUSED  )
+	PORT_BIT(0x02, IP_ACTIVE_LOW, IPT_UNUSED  )
+	PORT_BIT(0x04, IP_ACTIVE_LOW, IPT_UNUSED  ) 
+    PORT_BIT(0x10, IP_ACTIVE_LOW, IPT_BUTTON1) PORT_NAME("K3") // K0
+	PORT_BIT(0x20, IP_ACTIVE_LOW, IPT_BUTTON2) PORT_NAME("K2") // K1
+	PORT_BIT(0x40, IP_ACTIVE_LOW, IPT_BUTTON3) PORT_NAME("K1") // K2
+	PORT_BIT(0x80, IP_ACTIVE_LOW, IPT_BUTTON4) PORT_NAME("K0") // K3
+	
+	PORT_START("P1")
+
+    PORT_BIT(0x01, IP_ACTIVE_LOW, IPT_UNKNOWN )
+    PORT_BIT(0x02, IP_ACTIVE_LOW, IPT_UNKNOWN )
+    PORT_BIT(0x04, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT(0x08, IP_ACTIVE_LOW, IPT_UNKNOWN )
+    PORT_BIT(0x10, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT(0x20, IP_ACTIVE_LOW, IPT_CUSTOM) PORT_READ_LINE_DEVICE_MEMBER("hopper", FUNC(hopper_device::line_r))  // is from hopper? will cause error 02 During reset nvram
+    PORT_BIT(0x40, IP_ACTIVE_LOW, IPT_UNKNOWN ) 
+    PORT_BIT(0x80, IP_ACTIVE_LOW, IPT_UNKNOWN ) 
+
+
+
+PORT_START("DSW1")
+	PORT_DIPUNKNOWN_DIPLOC(0x01, 0x01, "SW1:1")
+	PORT_DIPUNKNOWN_DIPLOC(0x02, 0x02, "SW1:2")
+	PORT_DIPUNKNOWN_DIPLOC(0x04, 0x04, "SW1:3")
+	PORT_DIPUNKNOWN_DIPLOC(0x08, 0x08, "SW1:4")
+	PORT_DIPUNKNOWN_DIPLOC(0x10, 0x10, "SW1:5")
+	PORT_DIPUNKNOWN_DIPLOC(0x20, 0x20, "SW1:6")
+	PORT_DIPUNKNOWN_DIPLOC(0x40, 0x40, "SW1:7")
+	PORT_DIPUNKNOWN_DIPLOC(0x80, 0x80, "SW1:8")
+
+
+PORT_START("DSW2")
+	PORT_DIPUNKNOWN_DIPLOC(0x01, 0x01, "SW2:1")
+	PORT_DIPUNKNOWN_DIPLOC(0x02, 0x02, "SW2:2")
+	PORT_DIPUNKNOWN_DIPLOC(0x04, 0x04, "SW2:3")
+	PORT_DIPUNKNOWN_DIPLOC(0x08, 0x08, "SW2:4")
+INPUT_PORTS_END
+
+
+
+
+void fruitctk_state::p1_w(uint8_t data)
+{
+
+	m_hopper->motor_w(BIT(data, 3)); // ?? from guessed. Wrong
+    logerror("P1.0 Write to %02x\n", data);
+
+	
+}
+
+
+
+void fruitctk_state::multiplex_7seg_w(uint8_t data)
+{
+	m_selected_7seg_module = data;
+}
+
+uint8_t fruitctk_state::keyboard_r()
+{
+	switch (m_selected_7seg_module & 0x07)
+	{
+	case 0:
+	case 1:
+	case 2:
+	case 3:
+		return m_inputs[m_selected_7seg_module & 0x07]->read();
+	default:
+		return 0x00;
+	}

```

Comment:
What's the point of this switch-case? Is bit 2 just a selection between p1 and p2 sides?